src/Microsoft.ML.Data/Model/Onnx/OnnxContext.cs

```diff
@@ -49,6 +49,8 @@ public abstract class OnnxContext
         /// <param name="removeColumn">IDataView column to stop tracking</param>
         public abstract void RemoveVariable(string variableName, bool removeColumn);
 
+        public abstract string AddVariable(string colName);
```

Comment:
>public abstract string AddVariable(string colName); [](start = 8, length = 51)

Why are you making a private method part of the interface? If you need to add a variable you should use AddIntermediateVariable. This function is used internally by AddIntermediateVariable.

Reply:
In that case you should may be modify AddIntermediateVariable to include this functionality.

Reply:
Done.